src/pnpq/apt/protocol.py

```diff
@@ -958,3 +950,56 @@ class AptMessage_MGMSG_MOT_MOVE_STOPPED(AptMessageHeaderOnlyChanIdent):
 @dataclass(frozen=True, kw_only=True)
 class AptMessage_MGMSG_RESTOREFACTORYSETTINGS(AptMessageHeaderOnlyNoParams):
     message_id = AptMessageId.MGMSG_RESTOREFACTORYSETTINGS
+
+
+@dataclass(frozen=True, kw_only=True)
+class AptMessage_MGMSG_MOT_SET_EEPROMPARAMS(AptMessageWithData):
+    data_length: ClassVar[int] = 4
+    message_id = AptMessageId.MGMSG_MOT_SET_EEPROMPARAMS
+    message_struct: ClassVar[Struct] = Struct(
+        f"{AptMessageWithData.header_struct_str}{ATS.WORD}{ATS.WORD}"
+    )
+
+    chan_ident: ChanIdent
+    message_id_to_save: AptMessageId
+
+    @classmethod
+    def from_bytes(cls, raw: bytes) -> "AptMessage_MGMSG_MOT_SET_EEPROMPARAMS":
+        (
+            message_id,
+            data_length,
+            destination,
+            source,
+            chan_ident,
+            message_id_to_save,
+        ) = cls.message_struct.unpack(raw)
+
+        if message_id != cls.message_id:
+            raise ValueError(
+                f"Expected message ID {cls.message_id.value}, but received {message_id} instead. Full raw data was {raw!r}"
+            )
+        if data_length != cls.data_length:
+            raise ValueError(
+                f"Expected data packet length {cls.data_length}, but received {data_length} instead. Full raw data was {raw!r}"
+            )
+        if destination & 0x80 != 0x80:
+            raise ValueError(
+                f"Expected the destination's highest bit to be 1, indicating that a data packet follows, but it was 0. Full raw data was {raw!r}"
+            )
+
+        return AptMessage_MGMSG_MOT_SET_EEPROMPARAMS(
+            destination=Address(destination & 0x7F),
+            source=Address(source),
+            chan_ident=ChanIdent(chan_ident),
+            message_id_to_save=message_id_to_save,
```

Comment:
There's an ugly little bug here that should have been caught by type checking but wasn't. I think this is because I made the enum inherit from `int`. I'm rethinking that decision, but I can't change that as part of this PR.

```suggestion
            message_id_to_save=AptMessageId(message_id_to_save),
```

After making this change, one of your tests will start to fail.